orttraining/orttraining/test/python/orttraining_test_python_bindings.py

```diff
@@ -0,0 +1,243 @@
+            dynamic_axes[output_name].update({dim_idx: f"{output_name}_dim{dim_idx}"})
+
+    f = io.BytesIO()
+    torch.onnx.export(
+        torch_model,
+        model_inputs,
+        f,
+        input_names=input_names,
+        output_names=output_names,
+        opset_version=14,
+        do_constant_folding=False,
+        training=torch.onnx.TrainingMode.TRAINING,
+        dynamic_axes=dynamic_axes,
+        export_params=True,
+        keep_initializers_as_inputs=False,
+    )
+    return onnx.load_model_from_string(f.getvalue())
+
+
+def _create_training_models():
+    # Given
+    device = "cuda"
+    batch_size, input_size, hidden_size, output_size = 64, 784, 500, 10
+    _, onnx_model = _get_models(device, batch_size, input_size, hidden_size, output_size)
+
+    # Build the onnx model with loss
+    simple_model = SimpleModelWithCrossEntropyLoss()
+    with onnxblock.onnx_model(onnx_model) as accessor:
+        _ = simple_model(onnx_model.graph.output[0].name)
+        eval_model = accessor.eval_model
+
+    optimizer = onnxblock.optim.AdamW()
+    with onnxblock.onnx_model() as accessor:
+        _ = optimizer(simple_model.parameters())
+        optimizer_model = accessor.model
+
+    return simple_model, onnx_model, optimizer_model, eval_model
+
+
+def test_train_step():
+    # Initialize Models
+    simple_model, onnx_model, _, _ = _create_training_models()
+    trainable_params, non_trainable_params = simple_model.parameters()
+
+    # Generating random data for testing.
+    inputs = torch.randn(64, 784).numpy()
+    labels = torch.randint(high=10, size=(64,), dtype=torch.int32).numpy()
+    forward_inputs = OrtValueVector()
+    forward_inputs.reserve(2)
+    forward_inputs.push_back(OrtValue.ortvalue_from_numpy(inputs)._ortvalue)
+    forward_inputs.push_back(OrtValue.ortvalue_from_numpy(labels)._ortvalue)
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # Save models & checkpoint files to load them later.
+        checkpoint_file_path = os.path.join(temp_dir, "checkpoint")
+        onnxblock.save_checkpoint((trainable_params, non_trainable_params), checkpoint_file_path)
+
+        model_file_path = os.path.join(temp_dir, "training_model.onnx")
+        onnx.save(onnx_model, model_file_path)
+
+        # Create a Module.
+        model = Module(model_file_path, checkpoint_file_path)
+        model.train()
+        fetches = model(forward_inputs)
+        assert fetches
+
+
+def test_eval_step():
+    # Initialize Models
+    simple_model, onnx_model, _, eval_model = _create_training_models()
+    trainable_params, non_trainable_params = simple_model.parameters()
+
+    # Generating random data for testing.
+    inputs = torch.randn(64, 784).numpy()
+    labels = torch.randint(high=10, size=(64,), dtype=torch.int32).numpy()
+    forward_inputs = OrtValueVector()
+    forward_inputs.reserve(2)
+    forward_inputs.push_back(OrtValue.ortvalue_from_numpy(inputs)._ortvalue)
```

Comment:
Conversion from numpy to OrtValueVector can be extracted into a util mtd. Can we add this util mtd in our product code to make it easier for the users to go from numpy to OrtValueVector?

Reply:
added a TODO for the next PR.

Reply:
Do we need an OrtValueVector? Can we not use a python list of OrtValues? Let's tr to avoid any non-pythonic structs in the user interface if possible.